Updating interp codes for station points.

EricEngle-NOAA opened the pull request:
Updating all (non-spectral) interpolation codes to fully support interpolation from a supported source grid to stations points.

As of this commit, all existing tests pass and testing of interpolation to stations from grib2io are successful.

EricEngle-NOAA committed:
Updating interp codes for station points.

Updating all (non-spectral) interpolation codes to fully support
interpolation from a supported source grid to stations points.

As of this commit, all existing tests pass and testing of interpolation
to stations from grib2io are successful.

edwardhartnett commented:
Do we have any tests for this code?

EricEngle-NOAA commented:
I do not have tests for this at this time.  I *think* I should be able to add an interp to stations test inside tests/interp_mod_grib2.F90

AlexanderRichert-NOAA commented:
Hi @EricEngle-NOAA I just added tests for station points which are now in develop. Can you merge those and see how the tests go? With maybe the exception of neighbor-budget interpolation, I think the stations points functionality is generally working (though that's based on limited testing).

EricEngle-NOAA committed:
Merge branch 'NOAA-EMC:develop' into fix-interp-to-stations

EricEngle-NOAA committed:
Update interp_mod_grib2.F90

Fixing some code related to testing interpolation to station points.

EricEngle-NOAA committed:
Update for gdswzd_mod.F90

Added case for select type for "station point" grid and appropriate
logic to initially set XPTS and YPTS vectors.

EricEngle-NOAA commented:
The latest couple of commits have fixed the failing/crashing station points tests.

AlexanderRichert-NOAA commented:
That's great-- While we're at it, do you think it would be possible to enable the tests for the neighbor-budget scheme that are currently commented out?

EricEngle-NOAA commented:
Yes.  I will work on that tomorrow.

EricEngle-NOAA committed:
Adding station points tests for neighbor-budget.

EricEngle-NOAA commented:
The neighbor-budget station tests have been implemented and are working.

AlexanderRichert-NOAA commented on src/neighbor_budget_interp_mod.F90:
```diff
@@ -388,13 +398,16 @@ SUBROUTINE interpolate_neighbor_budget_vector(IPOPT,grid_in,grid_out, &
-       CALL GDSWZD(grid_out, 0,MO,FILL,XPTS,YPTS, &
-            RLON,RLAT,NO,CROT,SROT)
+    if(to_station_points) then
+       CALL GDSWZD(grid_out, 0,MO,FILL,XPTS,YPTS,RLON,RLAT,NO,CROT,SROT)
```
Does this line need to be in both parts of the if/else block or can it be pulled out in front?

EricEngle-NOAA replied:
Yes.  That can be pulled of the if block.  I will fix.

AlexanderRichert-NOAA commented on src/neighbor_interp_mod.F90:
```diff
@@ -93,6 +93,7 @@ module neighbor_interp_mod
   !> - 3    unrecognized output grid
   !>
   !> @author Mark Iredell @date 96-04-10  
+  !> @author Eric Enlgle @date 23-05-04  
```
name typo

EricEngle-NOAA replied:
Will fix.

AlexanderRichert-NOAA commented:
This looks great, thanks @EricEngle-NOAA. If you can check on and resolve my last two comments then I'm happy to get this merged.

EricEngle-NOAA committed:
Clean up of neighbor_budget_interp_mod.F90.
